code/client/launcher/Console.Logging.cpp

```diff
@@ -172,57 +177,85 @@ struct LoggerInit
 
 static LoggerInit logger;
 
-extern "C" DLL_EXPORT void AsyncTrace(const char* string)
+extern "C" DLL_EXPORT void AsyncTrace(const char* str)
 {
 	std::string threadName = GetThreadName();
 
-	g_logPrintQueue.push({ threadName, string });
+	g_logPrintQueue.push({ threadName, str });
 	g_logCondVar.notify_all();
 }
 
 void InitLogging()
 {
 	SetThreadName(-1, "MainThrd");
 
-	if (OpenMutex(SYNCHRONIZE, FALSE, L"CitizenFX_LogMutex_Mod") == nullptr)
+	const auto handle_deleter = [](HANDLE handle){ if (handle != INVALID_HANDLE_VALUE) CloseHandle(handle); };
+	using UniqueHandle = std::unique_ptr<std::remove_pointer<HANDLE>::type, decltype(handle_deleter)>;
+
+	UniqueHandle mutexHandle(
+		OpenMutex(SYNCHRONIZE, FALSE, L"CitizenFX_LogMutex_Mod"), 
+		handle_deleter
+	); // RAII exception safe wrapper of WinAPI HANDLE
+
+	if (NULL == mutexHandle)
 	{
 		// create the mutex
-		CreateMutex(nullptr, TRUE, L"CitizenFX_LogMutex_Mod");
-
-		CreateDirectory(MakeRelativeCitPath(L"logs/").c_str(), NULL);
+		mutexHandle.reset(CreateMutex(NULL, TRUE, L"CitizenFX_LogMutex_Mod"));
 
-		SYSTEMTIME curTime;
-		GetSystemTime(&curTime);
+		auto relativeCitPath = MakeRelativeCitPath(L"logs/");
 
-		for (auto& f : std::filesystem::directory_iterator{ MakeRelativeCitPath(L"logs/") })
+		if(0 == CreateDirectory(relativeCitPath.c_str(), NULL))
```

Comment:
`if(`? a tad nitpicky here, but could make sense

Reply:
Just following the docs from MS =)

> If the function fails, the return value is zero

Anyway, could we just replace ugly WinAPI with modern C++ filesystem variant a be done with it?